skimage/io/_plugins/matplotlib_plugin.py

```diff
@@ -169,10 +169,33 @@ def imshow_collection(ic, *args, **kwargs):
     """Display all images in the collection.
 
     """
-    fig, axes = plt.subplots(1, len(ic))
+    N = len(ic)
+    print N
+    k = (N * 12)**(0.5)
+    r1 = int(k/4)
```

Comment:
I think here I would prefer if you used `from math import floor` at the top of the file and then `r1 = floor(k/4)`. This would make it more explicit that it is rounding down.

Also, in the code below, could you add comments explaining the logic behind the maths? The paragraph you used in the GitHub issue would be sufficient (perhaps broken down over several lines). It will be very helpful to future maintainers!